FirebaseCore/Sources/FIRApp.m

```diff
@@ -871,14 +879,16 @@ - (void)subscribeForAppDidBecomeActiveNotifications {
   NSNotificationName notificationName = UIApplicationDidBecomeActiveNotification;
 #elif TARGET_OS_OSX
   NSNotificationName notificationName = NSApplicationDidBecomeActiveNotification;
+#elif TARGET_OS_WATCH_7_OR_LATER
+  NSNotificationName notificationName = WKApplicationDidBecomeActiveNotification;
 #endif
```

Comment:
Here add
#elif TARGET_OS_WATCH_6 or before?
[self logCoreTelemetryIfEnabled]

Reply:
I understand. I actually need to change the implementation to do runtime checks because build time checking for the OS version will not work as expected.

Reply:
@ncooke3 See https://github.com/google/GoogleUtilities/blob/main/GoogleUtilities/Environment/third_party/GULAppEnvironmentUtil.m#L245 for references on runtime version checking

Reply:
Thanks, I looked into it but ultimately used `if (@available(watchOS 7.0, *))` since it was more lightweight. LMK what you think...

Reply:
Agreed. LGTM